content/troubleshooting/ruby/Get-ruby-agent-logs.md

```diff
@@ -0,0 +1,111 @@
+
+* The web application is unresponsive after installing the Ruby agent.
+* The console log for the process has a Segmentation Fault. This will probably result in an error message containing
+`[BUG] Segmentation fault at: ` followed by a memory address.
+
+Once you confirm that the observed bug is a process crash, you're ready to gather information to file a bug.
+
+### Gather information on the process crash
+
+Complete the following steps to gather information to send to Contrast.
+
+* Add the following configuration to the YAML file to enable verbose logging and logging of every interaction between
+  the agent and the Contrast Service.
+
+```yaml
+
+agent:
+  logger:
+    level: DEBUG
+    path: ./contrast_agent_debug.log
+  service:
+    level: DEBUG
+    path: ./contrast_service_debug.log
+
+```
+
+* Verify that the Contrast-Service is no longer running. If the Service is still running in your environment, terminate
+  it manually.
+  * In Windows, you can run `tasklist /fi "imagename eq Contrast-Service.exe"`.
+  * In Linux or Mac, you can run `ps aux | grep Contrast-Service`.
+* Clear your console.
+* Exercise the application to reproduce the crash.
+
+Once you've reproduced the crash, gather the following items and include them in your bug report:
+
+* **Agent Logs:** The *contrast_agent_debug.log* and *contrast_service_debug.log* files; both files should be in the
+  application's root directory.
+* **Application Logs:** The logs from your application run, which are most likely in the *logs* directory of the application.
+* **Console Logs:** The results of the process crash in your terminal.
+* **State Logs:** While the agent logs capture much of the operating system information, it's helpful to know what,
+  if any, third-party Gems are being used. Even though it's not strictly a log, include your *Gemfile.lock* as part of the reproduction information collected.
+
+You should restore your logging levels to their original settings at the end of this process.
+
+## Unhandled Managed Exception
+
+### Verify an unhandled exception
+
+The above process also helps the Ruby agent team resolve issues such as application errors caused by the Ruby agent.
+Use the following indicators to determine if the Ruby agent is causing an application error.
+
+* You've observed the application working normally without the agent.
+* You've observed a page of the application "crashing" (returning a 500 error) under the agent.
+* There are no errors in the application logs in the *logs* directory.
+
+### Gather information on the exception
+
+Follow a similar process to gather agent logs to include with your bug report, omitting the **Console Logs** step.
+
+## Additional Logging
+
+In rare cases, additional information may be required to properly diagnose the root cause of the undesired behavior. To
+facilitate this, the Ruby agent provides a convenience wrapper around Ruby's built in Head Dump utility. To take a
```

Comment:
Is this supposed to be "heap" as well, or should I keep it as "head"?

Reply:
That should be `heap`. Sorry.